src/routes/safe/components/Settings/RemoveSafeModal/index.tsx

```diff
@@ -11,24 +11,59 @@ import Row from 'src/components/layout/Row'
 import PrefixedEthHashInfo from 'src/components/PrefixedEthHashInfo'
 import { currentSafeWithNames } from 'src/logic/safe/store/selectors'
 import removeSafe from 'src/logic/safe/store/actions/removeSafe'
-import { getExplorerInfo } from 'src/config'
+import { getChainById, getExplorerInfo } from 'src/config'
 import Col from 'src/components/layout/Col'
-import { WELCOME_ROUTE, history } from 'src/routes/routes'
+import { WELCOME_ROUTE, history, SafeRouteParams, generateSafeRoute, SAFE_ROUTES } from 'src/routes/routes'
+import useLocalSafes, { LocalSafes } from 'src/logic/safe/hooks/useLocalSafes'
+import { currentChainId } from 'src/logic/config/store/selectors'
+import { useMemo } from 'react'
+import { SafeRecordProps } from 'src/logic/safe/store/models/safe'
 
 type RemoveSafeModalProps = {
   isOpen: boolean
   onClose: () => void
 }
 
+function getNextAvailableSafe(currentChainId: string, currentSafeAddress: string, localSafes: LocalSafes) {
+  const availableSafes = Object.values(localSafes)
+    .flat()
+    .filter((safe) => safe.address !== currentSafeAddress)
+  const sameNetworkSafes = availableSafes.filter((safe) => safe.chainId === currentChainId)
+
+  if (sameNetworkSafes.length > 0) {
+    return sameNetworkSafes[0]
+  }
+
+  return availableSafes[0]
```

Comment:
So shall we remove this and redirect to welcome when removing the last Safe on a given network?